test/validation.test.ts

```diff
@@ -0,0 +1,586 @@
+        ],
+        ...schemaRule
+      }
+    ]
+  };
+  expect(validator(uischema)).toEqual([]);
+});
+
+test('validates group layout', () => {
+  expect(validator(uischemaGroup)).toEqual([]);
+});
+
+test('validates control element', () => {
+  expect(validator(uischemaControl)).toEqual([]);
+});
+
+test('validates SHOW effect of rule', () => {
+  const rule = {
+    rule: {
+      effect: 'SHOW',
+      condition: {
+        type: 'LEAF' ,
+        scope: '#/properties/alive',
+        expectedValue: false
+      }
+    }
+  };
+
+  const uischema = {...uischemaHorizontalLayout, ...rule };
+  expect(validator(uischema)).toEqual([]);
+});
+
+test('validates DISABLE effect of rule', () => {
+  const rule = {
+    rule: {
+      effect: 'DISABLE',
+      condition: {
+        type: 'LEAF' ,
+        scope: '#/properties/alive',
+        expectedValue: false
+      }
+    }
+  };
+
+  const uischema = {...uischemaHorizontalLayout, ...rule };
+  expect(validator(uischema)).toEqual([]);
+});
+
+test('validates ENABLE effect of rule', () => {
+  const rule = {
+    rule: {
+      effect: 'ENABLE',
+      condition: {
+        type: 'LEAF' ,
+        scope: '#/properties/alive',
+        expectedValue: false
+      }
+    }
+  };
+
+  const uischema = {...uischemaHorizontalLayout, ...rule };
+  expect(validator(uischema)).toEqual([]);
+});
+
+test('validates control element, label type boolean', () => {
+  const uischema = {
+    type: 'HorizontalLayout',
+    elements: [
+      {
+        type: 'Control',
+        label: true,
+        scope: '#/properties/occupation'
+      }
+    ]
+  };
+
+  expect(validator(uischema)).toEqual([]);
+});
+
+test('validates control element, label type string', () => {
+  expect(validator(uischemaControl)).toEqual([]);
+});
+
+test('validates control element, label type object', () => {
+  const uischema = {
+    type: 'HorizontalLayout',
+    elements: [
+      {
+        type: 'Control',
+        label: {
+          show: true,
+          text: 'Occupation'
+        },
+        scope: '#/properties/occupation'
+      }
+    ]
+  };
+
+  expect(validator(uischema)).toEqual([]);
+});
+
+test('invalid control element, invalid label', () => {
+  const uischema = {
+    type: 'HorizontalLayout',
+    elements: [
+      {
+        type: 'Control',
+        label: {
+          label: 'Occupation'
+        },
+        scope: '#/properties/occupation'
+      }
+    ]
+  };
+  const errors = validator(uischema);
+  expect(errors[0].message).toEqual('should be string');
+});
+
+test('invalid control element, label with additional properties', () => {
+  const uischema = {
+    type: 'HorizontalLayout',
+    elements: [
+      {
+        type: 'Control',
+        label: {
+          show: true,
+          text: 'Occupation',
+          hide: false
+        },
+        scope: '#/properties/occupation'
+      }
+    ]
+  };
+  const errors = validator(uischema);
+  expect(errors[0].message).toEqual('should be string');
```

Comment:
I'd rather use `toHaveLength` or a similar matcher; It's not immediately obvious why the message is `should be string`.